airflow/executors/celery_executor.py

```diff
@@ -188,14 +189,13 @@ def _process_tasks(self, task_tuples_to_send: List[TaskInstanceInCelery]) -> Non
         self.log.debug('Sent all tasks.')
 
         for key, _, result in key_and_async_results:
+            self.queued_tasks.pop(key)
             if isinstance(result, ExceptionWithTraceback):
                 self.log.error(  # pylint: disable=logging-not-lazy
                     CELERY_SEND_ERR_MSG_HEADER + ":%s\n%s\n", result.exception, result.traceback
                 )
+                self.event_buffer[key] = State.FAILED, None
```

Comment:
No hard opinion but for me this looks more readable
```suggestion
                self.event_buffer[key] = (State.FAILED, None)
```

Reply:
Agreed, and I normally do our this way. Wonder why I didn't this time